independent-projects/qute/core/src/main/java/io/quarkus/qute/IfSectionHelper.java

```diff
@@ -553,25 +568,22 @@ static Operator from(String value) {
         }
 
         static BigDecimal getDecimal(Object value) {
-            BigDecimal decimal;
-            if (value instanceof BigDecimal) {
-                decimal = (BigDecimal) value;
-            } else if (value instanceof BigInteger) {
-                decimal = new BigDecimal((BigInteger) value);
-            } else if (value instanceof Integer) {
-                decimal = new BigDecimal((Integer) value);
-            } else if (value instanceof Long) {
-                decimal = new BigDecimal((Long) value);
-            } else if (value instanceof Double) {
-                decimal = new BigDecimal((Double) value);
-            } else if (value instanceof Float) {
-                decimal = new BigDecimal((Float) value);
-            } else if (value instanceof String) {
-                decimal = new BigDecimal(value.toString());
-            } else {
-                throw new TemplateException("Not a valid number: " + value);
-            }
-            return decimal;
+            if (value instanceof BigDecimal decimal) {
+                return decimal;
+            } else if (value instanceof BigInteger bigInteger) {
+                return new BigDecimal(bigInteger);
+            } else if (value instanceof Integer integer) {
+                return new BigDecimal(integer);
```

Comment:
Use `BigDecimal.valueOf` to take advantage of the internal caching?

Reply:
Good point.